python/ribasim/ribasim/input_base.py

```diff
@@ -374,7 +378,13 @@ def _from_db(cls, path: Path, table: str):
         with closing(connect(path)) as connection:
             if exists(connection, table):
                 # pyogrio hardcodes fid name on reading
-                df = gpd.read_file(path, layer=table, fid_as_index=True)
+                df = pyogrio.read_dataframe(
+                    path,
+                    layer=table,
+                    fid_as_index=True,
+                    use_arrow=True,
+                    arrow_to_pandas_kwargs={"types_mapper": pd.ArrowDtype},
+                )
```

Comment:
Only pyogrio supported this directly, geopandas not yet.

Reply:
Best to place an inline comment to the specific kwarg that's not supported yet?

Reply:
Actually I misunderstood that all extra kwargs were passed from `gpd.read_file` to `pyogrio.read_dataframe` if `engine="pyogrio"`, so I went back to `gpd.read_file` and added a comment on `arrow_to_pandas_kwargs`, which ends up in `pyarrow`, not `pyogrio`.